scico/linop/xray/_axitom/backprojection.py

```diff
@@ -0,0 +1,171 @@
+"""
+This file is a modified version of "backprojection.py" from the
+[AXITOM](https://github.com/PolymerGuy/AXITOM) package.
+
+Filtered back projection functions
+
+This module contains the Feldkamp David Kress filtered back projection
+routines.
+"""
+
+from typing import Optional
+
+import jax.numpy as jnp
+from jax import Array
+from jax.scipy.ndimage import map_coordinates
+
+from .config import Config
+from .filtering import ramp_filter_and_weight
+from .utilities import rotate_coordinates
+
+
+def map_object_to_detector_coords(object_xs, object_ys, object_zs, config):
+    """Map the object coordinates to detector pixel coordinates
+    accounting for cone beam divergence.
+
+    Parameters
+    ----------
+    object_xs : np.ndarray
+        The x-coordinate array of the object to be reconstructed
+    object_ys : np.ndarray
+        The y-coordinate array of the object to be reconstructed
+    object_zs : np.ndarray
+        The z-coordinate array of the object to be reconstructed
+    config : obj
+        The config object containing all necessary settings for the
+        reconstruction
+
+    Returns
+    -------
+    detector_cords_a
+        The detector coordinates along the a-axis corresponding to the
+        given points
+    detector_cords_b
+        The detector coordinates along the b-axis corresponding to the
+        given points
+    """
+
+    detector_cords_a = (
+        ((object_ys * config.source_to_detector_dist) / (object_xs + config.source_to_object_dist))
+        - config.detector_us[0]
+    ) / config.pixel_size_u
+
+    if object_xs.ndim == 2:
+        detector_cords_b = (
+            (
+                (object_zs[jnp.newaxis, jnp.newaxis, :] * config.source_to_detector_dist)
+                / (object_xs[:, :, jnp.newaxis] + config.source_to_object_dist)
+            )
+            - config.detector_vs[0]
+        ) / config.pixel_size_v
+
+    elif object_xs.ndim == 1:
+        detector_cords_b = (
+            (
+                (object_zs[jnp.newaxis, :] * config.source_to_detector_dist)
+                / (object_xs[:, jnp.newaxis] + config.source_to_object_dist)
+            )
+            - config.detector_vs[0]
+        ) / config.pixel_size_v
+    else:
+        raise ValueError("Invalid dimensions on the object coordinates")
+
+    return detector_cords_a, detector_cords_b
+
+
+def _fdk_axisym(projection_filtered, config, angles):
```

Comment:
Maybe `_backproj_axisym` is a better name? The docstring is also misleading because it doesn't include filtering (as far as I can tell).

Reply:
True, but this name is from the original `axitom` code: addressing minor issues like this would be a slippery slope.

Reply:
Ah, very reasonable.